lib/matplotlib/pyplot.py

```diff
@@ -1199,7 +1200,27 @@ def subplot(*args, **kwargs):
 
         # add ax2 to the figure again
         plt.subplot(ax2)
+
+        # make the first axes "current" again
+        plt.subplot(221)
+
     """
+    proj_placeholder = object()
```

Comment:
Why is the placeholder object necessary? Why not `None`?

Reply:
Because the user can palusibly pass in `None`, the goal here is to detect if the user passed us _anyting_ .  @anntzer 's suggest of calling this `unset` is better.